### ISSUE TITLE
Ask for help. I can not train the same mIoU officially provided.

### ISSUE THREAD

**Issue by JimmyMa99** [open-mmlab/mmsegmentation#2456]
I trained DeeplabV3+ on Pascal Voc 2012 with the default config `deeplabv3plus_r101-d8_512x512_40k_voc12aug.py`.
However, I got the result: mIoU: 0.6035
I don't know why. I downloaded the checkpoint from [(https://github.com/open-mmlab/mmsegmentation/tree/master/configs/deeplabv3plus)] and I test it with the same config and the result was the same as it mentioned.

Here is my config.
`norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained='open-mmlab://resnet101_v1c',
    backbone=dict(
        type='ResNetV1c',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        dilations=(1, 1, 2, 4),
        strides=(1, 2, 1, 1),
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        contract_dilation=True),
    decode_head=dict(
        type='DepthwiseSeparableASPPHead',
        in_channels=2048,
        in_index=3,
        channels=512,
        dilations=(1, 12, 24, 36),
        c1_in_channels=256,
        c1_channels=48,
        dropout_ratio=0.1,
        num_classes=21,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    auxiliary_head=dict(
        type='FCNHead',
        in_channels=1024,
        in_index=2,
        channels=256,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        num_classes=21,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)),
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
dataset_type = 'PascalVOCDataset'
data_root = 'data/VOCdevkit/VOC2012'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations'),
    dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
    dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 512),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=4,
    workers_per_gpu=4,
    train=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir=['SegmentationClass', 'SegmentationClassAug'],
        split=[
            'ImageSets/Segmentation/train.txt',
            'ImageSets/Segmentation/aug.txt'
        ],
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations'),
            dict(type='Resize', img_scale=(2048, 512), ratio_range=(0.5, 2.0)),
            dict(type='RandomCrop', crop_size=(512, 512), cat_max_ratio=0.75),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='PascalVOCDataset',
        data_root='data/VOCdevkit/VOC2012',
        img_dir='JPEGImages',
        ann_dir='SegmentationClass',
        split='ImageSets/Segmentation/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(2048, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=50, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict()
lr_config = dict(policy='poly', power=0.9, min_lr=0.0001, by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=40000)
checkpoint_config = dict(by_epoch=False, interval=4000)
evaluation = dict(interval=4000, metric='mIoU', pre_eval=True)
work_dir = 'mzmseg/deeplabv3p_40k_vocaug'
gpu_ids = [0]
auto_resume = False
`
And here is the log.
[20230106_002856.log](https://github.com/open-mmlab/mmsegmentation/files/10357219/20230106_002856.log)
Help wanted.

**Comment by JimmyMa99** [comment]
I Re-upload the config as txt
[deeplabv3plus_r101-d8_512x512_40k_voc12aug.txt](https://github.com/open-mmlab/mmsegmentation/files/10357232/deeplabv3plus_r101-d8_512x512_40k_voc12aug.txt)

**Comment by xiexinch** [comment]
Hi @JimmyMa99,
We train the model with 4 GPUs, you might try to train with 4 GPUs or more.

**Comment by JimmyMa99** [comment]
> Hi @JimmyMa99, We train the model with 4 GPUs, you might try to train with 4 GPUs or more.

Thanks, I will try it again.
If there are any advances, I will continue to provide them under this issue.

**Comment by JimmyMa99** [comment]
I train deeplabV3+ with 4*3090 and the mIoU is 78.38, thank you.
Here is the log.
[20230106_153022.log](https://github.com/open-mmlab/mmsegmentation/files/10361741/20230106_153022.log)
[20230106_212621.log](https://github.com/open-mmlab/mmsegmentation/files/10361714/20230106_212621.log)

But I'm still curious why the training accuracy is related to the number of GPUs?

**Comment by JimmyMa99** [comment]
> 

I found ` samples_per_gpu=4, workers_per_gpu=4,` in the config, if I train my network with this config in one GPU, does that mean my batch size is only 4?

And if I set ` samples_per_gpu=8, workers_per_gpu=2,`, is it possible to use it for training my own model?

**Comment by xiexinch** [comment]
> > 
> 
> I found ` samples_per_gpu=4, workers_per_gpu=4,` in the config, if I train my network with this config in one GPU, does that mean my batch size is only 4?
> 
> And if I set ` samples_per_gpu=8, workers_per_gpu=2,`, is it possible to use it for training my own model?

Hi @JimmyMa99,
We have a detailed description in the documentation.
https://mmsegmentation.readthedocs.io/en/latest/train.html#train-a-model
https://mmsegmentation.readthedocs.io/en/latest/faq.html#how-to-know-the-number-of-gpus-needed-to-train-the-model

**Comment by JimmyMa99** [comment]
Ok, thanks a lot.
